### ISSUE TITLE
Possibly Errorenous Civ AI Policy Selection Logic

### ISSUE THREAD

**Issue by chris03-dev** [yairm210/Unciv#10857]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Game Version

4.9.15

### Describe the bug

Civ AIs seems to select the same policies if a mod is detected. Quite easy to trigger, and also possibly game-breaking if you have been relying on strategies that depend on Civ AI policy selection behavior. This mostly started as an attempt to find a cause for another [issue](https://github.com/ravignir/Brave-New-World/issues/15) in the mod I was using.

### Steps to Reproduce

1. Open the file manager of your choice
2. Go to the root Unciv directory, where you may find the executable binary and/or the JAR file
3. Go to the `mods/` directory if it already exists. If it didn't exist, create the `mods/` directory yourself.
4. Create a directory in the `mods/` directory.
5. Open the Unciv application.
6. Select "Start new game", and for best results:
  - Select your Civ as Spectator
  - Select your AI Civs as Egypt, Denmark, France (preference for Tradition policies), Rome, Russia, and Sweden (preference for Liberty policies)
  - Set the Game Speed to Quick
  - Set the map size to the smallest possible
7. Enable AutoPlay and watch the bug unravel in the Politics tab.

Note: If you have at least one mod installed, you might be already experiencing this issue. To fix this, put all your mods in a backup directory, and then remove all of the files and folders within the `mods/` directory (it is not necessary to remove the `mods/` directory itself).

### Screenshots

My configuration:
![image](https://github.com/yairm210/Unciv/assets/69199285/49375919-beb8-4f07-8029-c88c96f89550)

---

Running three tests of 50 turns each without a folder within the `mod/` folder:
![image](https://github.com/yairm210/Unciv/assets/69199285/7d87a96a-6ba9-4b88-9752-e09757b4e7e1)

Test 1:
![image](https://github.com/yairm210/Unciv/assets/69199285/a3828bbb-0ac7-48d9-b57e-725018bb0fec)

Test 2:
![image](https://github.com/yairm210/Unciv/assets/69199285/8019239f-76d9-4649-b15d-6915a75ab93e)

Test 3:
![image](https://github.com/yairm210/Unciv/assets/69199285/9fcfb675-ab15-499a-88e4-5cfe630d912b)

---

Now with a folder within the `mods/` folder:
![image](https://github.com/yairm210/Unciv/assets/69199285/7590d04f-db20-49ad-a928-110760ec4875)

Test 1:
![image](https://github.com/yairm210/Unciv/assets/69199285/07798cce-a323-401e-897e-9813f8c5fc5b)

Test 2:
![image](https://github.com/yairm210/Unciv/assets/69199285/5833aab9-1311-4c39-ae06-42b8253fdcd3)

Test 3:
![image](https://github.com/yairm210/Unciv/assets/69199285/1c3d2df8-d9c0-4b05-8383-0365c38ff824)

### Operating System

Windows

**Comment by chris03-dev** [comment]
> Slightly unclear: Did you select the empty bogus mod for that second set of tests or not?

In my tests, yes. But I do have to note that earlier observations involving this bug has happened with the mods I already installed.

**Comment by chris03-dev** [comment]
Alright, I tried to run a few more tests without a mod...
I think I might have overstated the role of a mod folder leading to this issue, since the so-called conditions I set are not actually necessary for the game to bug out like this. I tried a few tests without any mod, and it occurs all the same.

The only concrete information I could get out of this whole thing is that the Civ AI policy-selection behavior is consistent (i.e. chose Liberty or Tradition every single time) whenever I start a new game, that is, until I close the game itself. Whether the first 3 nations mentioned will have different policy selection behavior is now anyone's guess when I boot the game up again.

Now I only have one question... is this policy selection behavior supposed to be a bit random? Because from what I can gather from your earlier message, the selection process should be deterministic.

If so, it should be random on every new game, instead of being random on every Unciv startup. Otherwise, I think there is a bigger problem than I could try to debug on my own.

**Comment by chris03-dev** [comment]
Alright, I found out that one of the mods did break policy selection. I just made a fix by adding weights to certain victory types and now it works as expected, granted that the looming issue regarding random policy selection per game session still pops up sometimes.

Speaking of which, is policy selection already dictated for the entire session by the time the game is booted up, or when it enters the world screen for the first time? Or is it supposed to be calculated every time "New game" is selected?
